docker-compose.yml

```diff
@@ -0,0 +1,27 @@
+version: '3.8'
+
+services:
+  db:
+    image: postgres:13.1-alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_USER=smallboard_user
+      - POSTGRES_PASSWORD=smallboard_password
+      - POSTGRES_DB=smallboard
+  web:
+    build: .
+    command: python manage.py runserver 0.0.0.0:8000
+    volumes:
+      - .:/usr/src/smallboard/
+    ports:
+      - 8000:8000
+    environment:
+      - DATABASE_URL=postgres://smallboard_user:smallboard_password@db/smallboard
```

Comment:
Is it possible to define the variables user, pw, and db in one place and then refer to them later, instead of duplicating the values? I forget if YAML lets you do this.

Reply:
Seems like we can use extension fields or an env_file: https://stackoverflow.com/questions/36283908/re-using-environment-variables-in-docker-compose-yml

That way, `DATABASE_URL` can be constructed using the `POSTGRES_` variables and all of those variables can be defined globally to the compose file:

    POSTGRES_USER=smallboard_user
    POSTGRES_PASSWORD=smallboard_password
    POSTGRES_DB=smallboard
    DATABASE_URL="postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db/${POSTGRES_DB}"

I'd advocate for *either* storing all variables in the docker-compose file using extension fields or storing all variables in a `docker.env` file, but not both. The latter option seems a bit simpler in this case tbh.

Reply:
Hmm it doesn't look to me like you can use extension fields to construct the DATABASE_URL. 

I think the snippet you posted requires setting those env variables on the host machine, right? Not sure this is really worth it, would rather just keep the config simpler. I don't think the duplication is a big deal.

Reply:
Yeah, it seems YAML doesn't have quite the substitution support we needed.

Agreed it's not a big deal and we can just leave it as is for now.